content/blog/introducing-pulumi-import-command/index.md

```diff
@@ -0,0 +1,404 @@
+
+```bash
+pulumi import aws:s3/bucket:Bucket infra-logs company-infra-logs
+```
+
+This command instructs Pulumi to import the S3 bucket called `company-infra-logs` with the Pulumi resource name
+`infra-logs`. The import command will output as follows:
+
+{{< asciicast id="asciinema" >}}
+
+The user can then see the generated code they need to use as part of their application. I imported this resource into a
+TypeScript application:
+
+{{< chooser language "typescript,python,csharp,go" >}}
+
+{{< choosable language typescript >}}
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+
+const demo = new aws.s3.Bucket("infra-logs", {
+    acl: "private",
+    bucket: "company-infra-logs",
+    forceDestroy: false,
+}, {
+    protect: true,
+});
+```
+{{< /choosable >}}
+{{< choosable language python >}}
+```python
+import pulumi
+import pulumi_aws as aws
+
+demo = aws.s3.Bucket("infra-logs",
+    acl="private",
+    bucket="company-infra-logs",
+    force_destroy=False,
+    opts=ResourceOptions(protect=True))
+```
+{{< /choosable >}}
+{{< choosable language go >}}
+```go
+package main
+
+import (
+	"github.com/pulumi/pulumi-aws/sdk/v3/go/aws/s3"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		_, err := s3.NewBucket(ctx, "infra-logs", &s3.BucketArgs{
+			Acl:          pulumi.String("private"),
+			Bucket:       pulumi.String("company-infra-logs"),
+			ForceDestroy: pulumi.Bool(false),
+		}, pulumi.Protect(true))
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+}
+```
+{{< /choosable >}}
+{{< choosable language csharp >}}
+```csharp
+using Pulumi;
+using Aws = Pulumi.Aws;
+
+class MyStack : Stack
+{
+    public MyStack()
+    {
+        var demo = new Aws.S3.Bucket("infra-logs", new Aws.S3.BucketArgs
+        {
+            Acl = "private",
+            Bucket = "company-infra-logs",
+            ForceDestroy = false,
+        }, new CustomResourceOptions
+        {
+            Protect = true,
+        });
+    }
+
+}
+```
+{{< /choosable >}}
+{{< /chooser >}}
+
+This is valid Pulumi application code and can be added to your Pulumi application. Notice that the Pulumi resource is
+marked as `protect: true`. This means Pulumi will not delete that resource unless this protection attribute is removed
+from the resource. If this code is not added to your Pulumi application, then a Pulumi update will return the following:
+
+```bash
+Diagnostics:
+  pulumi:pulumi:Stack (import-example-dev):
+    error: preview failed
+
+  aws:s3:Bucket (demo):
+    error: Preview failed: refusing to delete protected resource 'urn:pulumi:dev::import-example::aws:s3/bucket:Bucket::infra-logs'
+```
+
+## Importing Multiple Resources
+
+Should a user need to import multiple resources to be managed by Pulumi, the CLI `import command` can handle this
+scenario for us. We can assemble all of the resources into a JSON document as follows:
```

Comment:
"This is helpful when using the `pulumi import` command as part of scripting larger bulk imports of cloud resource"